File: floor_sum/solve.py
import math

a, b, n = map(int, input().split())
limit = a // math.gcd(a, b)
acc = [0]
for i in range(1, limit+1):
  acc.append(acc[-1] + (b*i)%a)

sum_all = ((n*(n+1))//2) * b
sum_decimal = (n//limit) * acc[-1] + acc[n%limit]

sum_integer = (sum_all-sum_decimal) // a

print(sum_integer)

# ans = (limit*b//a) * limit * ((n//limit-1)*(n//limit)//2)

# tmp = 0
# for i in range(1, limit+1):
#   tmp += (b*i)//a

# ans += tmp * (n//limit)

# for i in range(1, (n%limit)+1):
#   ans += (b*i)//a+(limit*b//a) * (n//limit)

# print(ans)

# ans = 0
# for i in range(1, n+1):
#   ans += (b*i)//a

# print(ans)

File: floor_sum/solve.cpp
#include <bits/stdc++.h>
#include <atcoder/all>
using namespace std;
#define i64 int64_t
#define rep(n) for(int i = 0; i < (n); i++)

int main() {
  i64 a, b, n;
  cin >> a >> b >> n;
  i64 limit = a / gcd(a, b);
  vector<i64> acc(limit+1);
  rep (limit) {
    acc[i+1] = acc[i] + (b*(i+1))%a;
  }

  i64 sum_all = ((n*(n+1))/2) * b;
  i64 sum_decimal = (n/limit) * acc[limit] + acc[n%limit];

  i64 sum_integer = (sum_all-sum_decimal) / a;
  cout << sum_integer << endl;
  // i64 ans = 0;
  // rep (n) {
  //   ans += (b*(i+1)) / a;
  // }
  // cout << ans << endl;
}